client/modules/sheaf-core/src/theme.ts

```diff
@@ -0,0 +1,291 @@
+  border     = "var(--colcode-border)"    ,
+  accent     = "var(--colcode-accent)"    ,
+  selection  = "var(--colcode-selection)" ,
+  text       = "var(--colcode-content)"   ,
+  comment    = "var(--colcode-comment)"   ,
+  doc        = "var(--colcode-commentdoc)",
+  punct      = "var(--colcode-punct)"     ,
+  operator   = "var(--colcode-operator)"  ,
+  keyword    = "var(--colcode-keyword)"   ,
+  logical    = "var(--colcode-logical)"   ,
+  string     = "var(--colcode-string)"    ,
+  entity     = "var(--colcode-entity)"    ,
+  type       = "var(--colcode-type)"      ,
+  ident      = "var(--colcode-ident)"     ,
+  func       = "var(--colcode-function)"  ,
+  constant   = "var(--colcode-constant)"  ,
+  property   = "var(--colcode-property)"  ,
+  tag        = "var(--colcode-tag)"       ,
+  classes    = "var(--colcode-class)"     ,
+  attr       = "var(--colcode-attribute)" ,
+  markup     = "var(--colcode-markup)"    ,
+  link       = "var(--colcode-link)"      ,
+  invalid    = "var(--colcode-invalid)"   ,
+  inserted   = "var(--colcode-inserted)"  ,
+  changed    = "var(--colcode-changed)"   ,
+  important  = "var(--colcode-important)" ,
+  highlight  = "var(--colcode-highlight)" ,
+  note       = "var(--colcode-note)"      ,
+  special    = "var(--colcode-special)"
+
+const confinementTheme = EditorView.theme(
+  {
+    "&": {
+      color: text,
+      backgroundColor: background,
+      "& ::selection": { backgroundColor: selection },
+      caretColor: accent,
+      "&.cm-focused": { outline: "none" },
+      width: "auto",
+      height: "100%"
+    },
+
+    ".cm-scroller": {
+      fontFamily: "var(--font-mono)",
+      height: "100%",
+      fontSize: "13px",
+      fontWeight: "400",
+      position: "relative",
+      overflowX: "auto",
+      zIndex: 0
+    },
+
+    ".cm-content": {
+      paddingBottom: "70vh",
+      maxWidth: "45rem",
+      lineHeight: "18px"
+    },
+
+    "&.cm-focused .cm-cursor": {
+      borderLeftColor: accent,
+      transition: "left 0.05s ease-out, top 0.05s ease-out"
+    },
+
+    "&.cm-focused .cm-cursorLayer": {
+      animation: "cubic-bezier(0.95, 0, 0.05, 1) cm-blink 1.2s infinite"
+    },
+
+    "@keyframes cm-blink": { "0%": {}, "50%": { opacity: "0" }, "100%": {} },
+    "@keyframes cm-blink2": { "0%": {}, "50%": { opacity: "0" }, "100%": {} },
+
+    "&.cm-focused .cm-selectionBackground": { backgroundColor: selection },
+    ".cm-selectionBackground": { backgroundColor: selection },
+    ".cm-activeLine": { background: hover },
+    ".cm-selectionMatch": { backgroundColor: selection },
+    ".cm-searchMatch": {
+      backgroundColor: selection,
+      borderRadius: "0.125rem"
+    },
+    ".cm-searchMatch.selected": {
+      backgroundColor: selection,
+      boxShadow: `0 0 0 0.075rem ${accent}`
+    },
+
+    ".cm-line": {
+      "& ::selection": { color: "inherit !important" },
+      "&::selection": { color: "inherit !important" }
```

Comment:
These are different?

Reply:
One is "if parent is selected" and another is "if something in the parent is selected"